superset/assets/src/chart/Chart.jsx

```diff
@@ -176,7 +176,7 @@ class Chart extends React.PureComponent {
       >
         {this.renderTooltip()}
 
-        {isLoading && <Loading size={50} />}
+        {chartStatus !== 'rendered' && <Loading size={50} />}
```

Comment:
How about updating the definition of `isLoading` on line 164?

**from**

```js
const isLoading = chartStatus === 'loading';
```

**to**

```js
const isLoading = chartStatus !== 'rendered';
```

Reply:
That's what I tried at first, but that broke things, maybe because of the css class `is-loading`?

Reply:
Variable-naming-wise `isLoading = chartStatus === 'loading'` makes more sense than `isLoading = chartStatus !== 'rendered'`, so I want to stick with it.

The confusion is more around the component being called `<Loading />` which should be shown while the component is still either loading (async call) or rendering (waiting for browser's attention).

Reply:
you could possibly add a new variable that makes this more readable `const showLoader = chartStatus !== 'rendered'`.

This is an interesting observation tho, I think it just shows that the chart rendering is actually really slow, esp for certain vis types and when you are loading a dashboard because there are a HUGE amount of DOM changes/additions. Theoretically the rendering is starting at the same time we were previously hiding the `Loader` ...